### PR TITLE
[CARBONDATA-3424] Carbon should throw proper exception when Select query with average function for substring of binary column

### PR DESCRIPTION
**Pull request by xubo245**
Carbon should throw proper exception when Select query with average function for substring of binary column

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 NO
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
      Yes
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
JIRA-3336

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3484/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3687/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11752/

File: integration/spark-common-test/src/test/resources/2000_testCSV.csv [hunk 1]
```diff
@@ -0,0 +1,18 @@
+,,,,,,,,,,,
```
Comment (KanakaKumar) [inline integration/spark-common-test/src/test/resources/2000_testCSV.csv]: Is there any particular reason to add a new CSV file ? If not, can you use already existing CSV file ?

**Comment by KanakaKumar** [comment]
LGTM

File: integration/spark-common-test/src/test/resources/2000_testCSV.csv (hunk 1 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/resources/2000_testCSV.csv]: yes, there issue is reported by tester. He use this csv file, it's difficult to replace it. I need reproduce it and validate with this file.

Comment (KanakaKumar) [reply integration/spark-common-test/src/test/resources/2000_testCSV.csv]: Please avoid adding files if not required. We can reproduce the root cause with any sample data.

Comment (xubo245) [reply integration/spark-common-test/src/test/resources/2000_testCSV.csv]: done

Comment (xubo245) [reply integration/spark-common-test/src/test/resources/2000_testCSV.csv]: removed

**Comment by KanakaKumar** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3713/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11779/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3508/

**Comment by xubo245** [comment]
@KanakaKumar CI pass, please check.

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**